R/matrix_form.R

```diff
@@ -525,7 +607,7 @@ mf_col_widths <- function(mf) {
       "number of columns in strings matrix (", NCOL(mf_strings(mf)), ")."
     )
   }
-  mf$col_widths <- value
+  mf <- mf_update_cinfo(mf, colwidths = value, rep_cols = NULL)
```

Comment:
I need to understand exactly what is the status of cinfo because I have troubles understanding when it is to be used instead of matrix_form entries

Reply:
cinfo (somewhat unfortunate naming on my part, though not worth changing I think) is the column data frame. its used for horizontal pagination of the mpf the same way rinfo is used for vertical pagination.

the pagination data.frame needs to be adjusted when column widths are set/changed, because it changes the self_extent column used in horiz pagination.

Reply:
I see! I had to fiddle a bit with self_extent for {rlistings} when wrapping keycolumns. You will see in paginate_mpfs something regarding that. Do you think that kind of behavior should rely on cinfo or rinfo? Because at the moment I based it on rinfo